### PR TITLE
Improve symbol api doc

### PR DESCRIPTION
**Pull request by kevinthesun**
pow
maximum
minimum
hypot
debug_str

### CODE REVIEW

File: python/mxnet/symbol.py [hunk 1]
```diff
@@ -1639,16 +1720,35 @@ def maximum(left, right):
...
+    2
+    >>> x = mx.sym.Variable('x')
+    >>> y = mx.sym.Variable('y')
+    >>> z = mx.sym.minimum(x, 4)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([3,5,2,10])})
+    >>> texec.forward()[0].asnumpy()
+    array([ 3.,  4.,  2.,  4.], dtype=float32)
+    >>> z = mx.sym.minimum(x, y)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([3,4]), 'y':mx.nd.array([10, 2])})
+    >>> texec.forward()[0].asnumpy()
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: use eval instead.

File: python/mxnet/symbol.py [hunk 2]
```diff
@@ -1006,12 +1006,50 @@ def _infer_shape_impl(self, partial, *args, **kwargs):
             # pylint: enable=too-many-locals
 
     def debug_str(self):
-        """Gets a debug string.
+        """Gets a debug string of symbol. It contains Symbol output, variables and
```
Comment (mli) [inline python/mxnet/symbol.py]: add an empty line before `It contains`

File: python/mxnet/symbol.py [hunk 3]
```diff
@@ -1006,12 +1006,50 @@ def _infer_shape_impl(self, partial, *args, **kwargs):
...
+        operators in the computation graph with their inputs, variables and attributes.
 
         Returns
         -------
-        debug_str : string
+        String
```
Comment (mli) [inline python/mxnet/symbol.py]: change to string

File: python/mxnet/symbol.py [hunk 4]
```diff
@@ -1006,12 +1006,50 @@ def _infer_shape_impl(self, partial, *args, **kwargs):
...
             Debug string of the symbol.
+
+        Examples
+        --------
+        >>> a = mx.sym.Variable('a')
+        >>> b = mx.sym.sin(a)
+        >>> c = 2 * a + b
+        >>> d = mx.sym.FullyConnected(data=c, num_hidden=10)
+        >>> d,debug_str()
```
Comment (mli) [inline python/mxnet/symbol.py]: typo? why there is a `,` after d?

File: python/mxnet/symbol.py [hunk 5]
```diff
@@ -1587,16 +1625,40 @@ def load_json(json_str):
...
+        The bases in x raised to the exponents in y.
+
+    Examples
+    --------
+    >>> mx.sym.pow(2, 3)
+    8
+    >>> x = mx.sym.Variable('x')
+    >>> y = mx.sym.Variable('y')
+    >>> z = mx.sym.pow(x, 2)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([1,2])})
```
Comment (mli) [inline python/mxnet/symbol.py]: use `z.eval` instead of bind and forward

File: python/mxnet/symbol.py [hunk 6]
```diff
@@ -1613,16 +1675,35 @@ def pow(base, exp):
 # pylint: disable=no-member
 # pylint: disable=redefined-builtin
 def maximum(left, right):
-    """ maximum left and right
+    """ Returns element-wise maximum of the input elements.
+    Both inputs can be Symbol or scalar number. Broadcasting is not supported.
```
Comment (mli) [inline python/mxnet/symbol.py]: add an empty line

File: python/mxnet/symbol.py [hunk 7]
```diff
@@ -1613,16 +1675,35 @@ def pow(base, exp):
...
+        The element-wise maximum of the input symbols.
+
+    Examples
+    --------
+    >>> mx.sym.maximum(2, 3.5)
+    3.5
+    >>> x = mx.sym.Variable('x')
+    >>> y = mx.sym.Variable('y')
+    >>> z = mx.sym.maximum(x, 4)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([3,5,2,10])})
```
Comment (mli) [inline python/mxnet/symbol.py]: use `z.eval`

File: python/mxnet/symbol.py [hunk 8]
```diff
@@ -1639,16 +1720,35 @@ def maximum(left, right):
 # pylint: disable=no-member
 # pylint: disable=redefined-builtin
 def minimum(left, right):
-    """ minimum left and right
+    """ Returns element-wise minimum of the input elements.
+    Both inputs can be Symbol or scalar number. Broadcasting is not supported.
```
Comment (mli) [inline python/mxnet/symbol.py]: empty line

File: python/mxnet/symbol.py [hunk 9]
```diff
@@ -1639,16 +1720,35 @@ def maximum(left, right):
...
+        The element-wise minimum of the input symbols.
+
+    Examples
+    --------
+    >>> mx.sym.minimum(2, 3.5)
+    2
+    >>> x = mx.sym.Variable('x')
+    >>> y = mx.sym.Variable('y')
+    >>> z = mx.sym.minimum(x, 4)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([3,5,2,10])})
```
Comment (mli) [inline python/mxnet/symbol.py]: use z.eval

File: python/mxnet/symbol.py [hunk 10]
```diff
@@ -1657,24 +1757,44 @@ def minimum(left, right):
...
+        return left if left < right else right
     else:
         raise TypeError('types (%s, %s) not supported' % (str(type(left)), str(type(right))))
 
 
 # pylint: disable=no-member
 # pylint: disable=redefined-builtin
 def hypot(left, right):
-    """ minimum left and right
+    """ Given the "legs" of a right triangle, return its hypotenuse.
```
Comment (mli) [inline python/mxnet/symbol.py]: no space before Given and an empty line after

File: python/mxnet/symbol.py [hunk 11]
```diff
@@ -1657,24 +1757,44 @@ def minimum(left, right):
...
+    Equivalent to sqrt(left**2 + right**2), element-wise.
```
Comment (mli) [inline python/mxnet/symbol.py]: add `` before sqrt and after )

File: python/mxnet/symbol.py [hunk 12]
```diff
@@ -1657,24 +1757,44 @@ def minimum(left, right):
...
+        The hypotenuse of the triangle(s)
+
+    Examples
+    --------
+    >>> mx.sym.hypot(3, 4)
+    5.0
+    >>> x = mx.sym.Variable('x')
+    >>> y = mx.sym.Variable('y')
+    >>> z = mx.sym.hypot(x, 4)
+    >>> texec = z.bind(mx.cpu(), {'x': mx.nd.array([3,5,2])})
```
Comment (mli) [inline python/mxnet/symbol.py]: use z.eval

File: python/mxnet/symbol.py (hunk 4 above)
Comment (kevinthesun) [inline python/mxnet/symbol.py]: This is weird since in local file this is a dot instead of comma.

**Comment by madjam** [comment]
@kevinthesun did you address all of Mu's feedback?

File: python/mxnet/symbol.py [hunk 13]
```diff
@@ -1617,16 +1657,38 @@ def load_json(json_str):
...
+    """Returns element-wise result of base element raised to powers from exp element.
+
+    Both inputs can be Symbol or scalar number.
+    Broadcasting is not supported. Use `broadcast_pow` instead.
 
     Parameters
     ---------
-    base: Symbol or Number
-    exp: Symbol or Number
+    base : Symbol or scalar
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: I think number is better here.

**Pull request merged**